sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -145,14 +147,18 @@ object AggregateUtils {
         GpuBindReferences.bindGpuReferences(updateExpressionsSeq(modeIndex), aggBufferAttributes)
             .asInstanceOf[Seq[CudfAggregate]]
       } else {
-        GpuBindReferences.bindGpuReferences(mergeExpressionsSeq(modeIndex), aggBufferAttributes)
+        GpuBindReferences.bindGpuReferences(mergeExpressionsSeq(modeIndex), mergeBufferAttributes)
             .asInstanceOf[Seq[CudfAggregate]]
       }
-      (expr.mode, cudfAggregates)
+      BoundCudfAggregate(expr, cudfAggregates)
     }
   }
 }
 
+case class BoundCudfAggregate(
+    aggExpression: GpuAggregateExpression,
+    boundCudfAggregate: Seq[CudfAggregate])
```

Comment:
Could we add some docs describing the requirements/guarantees associated with each of these? Like is CudfAggregate just the aggregate with GpuBoundReference as the input?  Why is this a sequence? Will there be one column of output for each aggregate?

Reply:
done.